tests/test_correlator.py

```diff
@@ -0,0 +1,159 @@
+from pathlib import Path
+
+from ase import Atoms
+from ase.io import read
+from ase.units import bar
+import numpy as np
+from pytest import approx
+from typer.testing import CliRunner
+from yaml import Loader, load
+
+from janus_core.calculations.md import NVE
+from janus_core.calculations.single_point import SinglePoint
+from janus_core.helpers.correlator import Correlator
+
+DATA_PATH = Path(__file__).parent / "data"
+MODEL_PATH = Path(__file__).parent / "models" / "mace_mp_small.model"
+runner = CliRunner()
+
+
+# pylint: disable=invalid-name
+def correlate(
+    x: Iterable[float], y: Iterable[float], *, fft: bool = True
+) -> Iterable[float]:
+    """
+    Direct correlation of x and y. If fft uses np.correlate in full mode.
+    """
+    n = min(len(x), len(y))
+    if fft:
+        cor = np.correlate(x, y, "full")
+        cor = cor[len(cor) // 2 :]
+        return cor / (n - np.arange(n))
+    cor = np.zeros(n)
+    for j in range(n):
+        for i in range(n - j):
+            cor[j] += x[i] * y[i + j]
+        cor[j] /= n - j
+    return cor
+
+
+def test_setup():
+    """Test initial values"""
+    cor = Correlator(blocks=1, points=100, averaging=2)
+    correlation, lags = cor.get()
+    assert len(correlation) == len(lags)
+    assert len(correlation) == 0
+
+
+def test_correlation():
+    """Test Correlator against np.correlate"""
+    points = 100
+    cor = Correlator(blocks=1, points=points, averaging=1)
+    signal = np.exp(-np.linspace(0.0, 1.0, points))
+    for val in signal:
+        cor.update(val, val)
+    correlation, lags = cor.get()
+
+    direct = correlate(signal, signal, fft=False)
+    fft = correlate(signal, signal, fft=True)
+
+    assert len(correlation) == len(lags)
+    assert all(lags == range(points))
+    assert direct == approx(correlation, rel=1e-10)
+    assert fft == approx(correlation, rel=1e-10)
+
+
+# pylint: disable=too-many-locals
+def test_md_correlations(tmp_path):
+    """Test correlations as part of MD cycle."""
+    file_prefix = tmp_path / "Cl4Na4-nve-T300.0"
+    traj_path = tmp_path / "Cl4Na4-nve-T300.0-traj.xyz"
+    stats_path = tmp_path / "Cl4Na4-nve-T300.0-stats.dat"
+    cor_path = tmp_path / "Cl4Na4-nve-T300.0-cor.dat"
+
+    single_point = SinglePoint(
+        struct_path=DATA_PATH / "NaCl.cif",
+        architecture="mace",
+        calc_kwargs={"model": MODEL_PATH},
+    )
+
+    # pylint: disable=unused-argument
+    def user_observable_a(atoms: Atoms, kappa, **kwargs) -> float:
+        """User specified getter for correlation"""
+        return (
+            kwargs["gamma"]
+            * kappa
+            * atoms.get_stress(include_ideal_gas=True, voigt=True)[-1]
+            / bar
+        )
+
+    def user_observable_b(atoms: Atoms) -> float:
+        """User specified getter for correlation"""
+        return atoms.get_stress(include_ideal_gas=True, voigt=True)[-1] / bar
+
+    nve = NVE(
+        struct=single_point.struct,
+        temp=300.0,
+        steps=100,
```

Comment:
Do you need 100 steps to get a useful correlation? It would be nice if it were a little shorter if possible, but it's fine if the results are not distinguishable enough otherwise for the tolerance.

Reply:
Given the test is now seeded we could choose say 10 steps for speed. For actual use probably more is needed for science, but this is just a test.

Originally it was because being unseeded the error value was unstable at << 100 steps.